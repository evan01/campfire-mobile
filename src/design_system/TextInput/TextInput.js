// @flow
import React from "react";
import { TextInput, StyleSheet } from "react-native";
import colors from "../styles/colors";
import fonts from "../styles/fonts";

const DefaultTextInput = (props) => <TextInput {...props} style={styles} />;

DefaultTextInput.displayName = "TextInput";

const styles = StyleSheet.flatten({
  height: 50,
  backgroundColor: colors.WHITE,
  borderColor: colors.BLACK,
  borderRadius: 25,
  marginLeft: 15,
  fontFamily: fonts.ROBOTO.THIN,
  fontSize: fonts.SIZE.FONT_SIZE_H2,
  paddingLeft: 20,
  padding: 14,
});
export default DefaultTextInput;
