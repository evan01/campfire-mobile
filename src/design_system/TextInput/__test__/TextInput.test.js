// @flow
import React from "react";
import renderer from "react-test-renderer";
import TextInput from "../TextInput";

describe("TextInput", () => {
  test("renders correctly", () => {
    const input = renderer.create(<TextInput name="bug-report" />);
    expect(input).toMatchSnapshot();
  });
});
