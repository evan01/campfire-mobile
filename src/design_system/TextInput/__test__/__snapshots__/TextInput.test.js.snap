// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  name="bug-report"
  rejectResponderTermination={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#000000",
      "borderRadius": 25,
      "fontFamily": "Roboto-Thin",
      "fontSize": 22,
      "height": 50,
      "marginLeft": 15,
      "padding": 14,
      "paddingLeft": 20,
    }
  }
  underlineColorAndroid="transparent"
/>
`;
